src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +162,171 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+                msgBuilder.Append($"Transactional ID '{TransactionalId}' : ");
+                if (validTransactionWriteAcl == false)
+                    msgBuilder.Append("WRITE operation ");
+                if (validTransactionDescribeAcl == false)
+                    msgBuilder.Append("DESCRIBE operation");
+                msgBuilder.Append(Environment.NewLine);
             }
 
-            throw new QueueLoadException(ex.Message, ex);
+            
+            if (validTopicWriteAcl == false)
+                msgBuilder.Append($"Topic '{topicName}' : WRITE operation");
+
+            message = msgBuilder.ToString();
+
+        }
+        catch (Exception)
+        {
+            return false;
         }
 
-        return count;
+        return true;
+    }
+
+    private static bool AssertAclMatchingName(string expectedName, AclBinding aclBinding)
+    {
+        return aclBinding.Pattern.ResourcePatternType switch
+        {
+            ResourcePatternType.Literal => aclBinding.Pattern.Name == expectedName,
+            ResourcePatternType.Prefixed => expectedName.StartsWith(aclBinding.Pattern.Name),
+            _ => throw new NotSupportedException()
+        };
+    }
+    
+
+    private void ThrowTransactionalIdAclAuthorizationException(Exception e)
+    {
+        string msg = $"ETL process: {Name}. Failed to initialize transactions for the producer instance. " +
+                     $"Add an required ACL permissions to your API Key that will make WRITE and DESCRIBE operations possible for transactional ID '{TransactionalId}'. ";
+                                    
+        if (Logger.IsOperationsEnabled)
+            Logger.Operations(msg, e);
+        // no need to abort transaction yet, this method is called when _producer.InitTransactions() fails
+        throw new QueueLoadException(msg, e);
+    }
+
+    private void ThrowTopicAclAuthorizationException(Exception e, string currentTopicName)
+    {
+        var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to initialize transactions for the producer instance.{Environment.NewLine}" +
+                                                    $"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopicName}' topic possible.");
+        AbortTransactionAfterException(errorMessageBuilder);
+        throw new QueueLoadException(errorMessageBuilder.ToString(), e);
+    }
+
+    private void AbortTransactionAfterException(StringBuilder errorMessageBuilder)
+    {
+        try
+        {
+            errorMessageBuilder.Append($"{Environment.NewLine}Aborting the transaction. ");
+            _producer.AbortTransaction();
+        }
+        catch (Exception ex)
+        {
+            errorMessageBuilder.Append("Aborting Kafka transaction failed too.");
+            throw new QueueLoadException(errorMessageBuilder.ToString(), ex);
```

Comment:
Please note that you might cover the actual error here. Since you're passing the exception from the abort operation but `AbortTransactionAfterException` is called inside catch blocks where the actual errors are handled

Reply:
solved

Reply:
https://github.com/ravendb/ravendb/pull/17577/commits/4480ef6e0448e40e1e6469b929f2397b82a35dca#diff-394718a1f7369d005e626a6b38815db2e0878a806221b3effe02a0cd3832746aR201-R203